src/components/views/settings/shared/SettingsSubsection.tsx

```diff
@@ -25,6 +26,10 @@ export interface SettingsSubsectionProps extends HTMLAttributes<HTMLDivElement>
     children?: React.ReactNode;
     // when true content will be justify-items: stretch, which will make items within the section stretch to full width.
     stretchContent?: boolean;
+    /*
+     * When true, the new UI style will be applied to the subsection.
+     */
+    newUi?: boolean;
```

Comment:
I generally like to call old designs "legacy" rather than calling their replacements "new", because "legacy" ages better. If this code still exists multiple years from now, it's much more likely that the old UI style will still be considered "legacy", than that the new UI style will still be considered "new". So, if it's easy to invert this flag and turn it into something like `legacy?: boolean`, I think that could help with its longevity.